File: src/main/java/nz/co/searchwellington/model/frontend/FrontendTag.java
package nz.co.searchwellington.model.frontend;

public class FrontendTag {
	
	private final String id, name;
	
	public FrontendTag(String id, String name) {
		this.id = id;
		this.name = name;
	}
	
	public String getId() {
		return id;
	}
	
	public String getName() {
		return name;
	}
	
	@Override
	public String toString() {
		return "FrontendTag [id=" + id + ", name=" + name + "]";
	}
	
}

File: src/main/scala/nz.co.searchwellington.model.frontend/FrontendTag.scala
package nz.co.searchwellington.model.frontend

case class FrontendTag(id: String, name: String, displayName: String, description: String, children: Seq[FrontendTag] = Seq.empty) {

  def getId: String = id

  def getDisplayName: String = displayName

  def getName: String = name

  def getDescription: String = description

  def getChildren: java.util.List[FrontendTag] = {
    import scala.collection.JavaConverters._
    children.asJava
  }

}